core/config/v2/types.go

```diff
@@ -110,6 +111,15 @@ func (c *Core) SetFrom(f *Core) {
 	c.Insecure.setFrom(&f.Insecure)
 }
 
+func (c *Core) ValidateConfig() (err error) {
+	_, verr := parse.HomeDir(*c.RootDir)
+	if err != nil {
+		err = multierr.Append(verr, ErrInvalid{Name: "RootDir", Value: true, Msg: "Failed to expand RootDir. Please use an explicit path."})
```

Comment:
If we are going to include the `verr` then we should wrap it, instead of just having them alongside each other. The config errors don't have that built-in, but we could do something like this:
```suggestion
		err = multierr.Append(err, ErrInvalid{Name: "RootDir", Value: true, Msg: fmt.Sprintf("Failed to expand RootDir. Please use an explicit path: %s", verr)})
```